src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_constructs/adf_codebuild.py

```diff
@@ -159,6 +159,17 @@ def determine_build_spec(codebuild_id, default_props, target=None):
             ),
         )
 
+    @staticmethod
+    def get_image_by_name(specific_image: str):
+        if hasattr(_codebuild.LinuxBuildImage,
+            (specific_image or DEFAULT_CODEBUILD_IMAGE).upper()):
+            return getattr(_codebuild.LinuxBuildImage,
+                (specific_image or DEFAULT_CODEBUILD_IMAGE).upper())
+        if specific_image.startswith('docker-hub://'):
+            specific_image = specific_image.split('docker-hub://')[-1]
+            return _codebuild.LinuxBuildImage.from_docker_registry(specific_image)
+        return specific_image
```

Comment:
Could you make it raise an error when we hit this line instead?

At the moment, it would return the CodeBuild image in both other return types.
While it returns a string when it is unable to find/match.

You can raise an exception that states that the CodeBuild image could not be found.
Please include the specific_image in the error message so it will be easy to debug.

Reply:
Good lord I was tired yesterday 😅 
Will something like this raise exception work for you?